src/main/com/scalyr/api/query/QueryService.java

```
@@ -482,6 +615,23 @@ private void checkResponseStatus(JSONObject rawApiResponse) throws ScalyrServerE
       this.continuationToken = continuationToken;
     }
 
+
+    /**
+     * Merge `a` and `b` into a new result (whose `b` matches follow those of `a`, and using b's continuationToken).
+     * Make sure to provide inputs in your desired order. May return one of its inputs if the other is null.
+     */
+    static LogQueryResult merge(LogQueryResult a, LogQueryResult b) {
+      if (a == null) return b;
+      if (b == null) return a;
+      LogQueryResult ret = new LogQueryResult(a.executionTime + b.executionTime, b.continuationToken);
+      ret.matches.addAll(a.matches);
+      ret.matches.addAll(b.matches);
```

Don't we need to sum the values here, rather than concatenate them? ...of course that's only valid if `function` is `count` and `buckets` is 1. Probably, if chunking is used, we should throw an exception for any parameters other than `function='count' buckets=1`.

---

We should chat about this.  Honoring `buckets` across chunks is a PITA - one of the chunks will be mis-sized, so we'd need to do some hackery to try to split N buckets across M chunks "correctly".

I've opted instead for "if you ask for `maxCount=100`, you will get up to 100 results PER CHUNK and we'll stop as soon as we hit one chunk that has > 100 matches.  Likewise, if you ask for `buckets = 10`, you will get 10 buckets PER CHUNK".

Actually doing this right is way more work than we want to invest.

---

Did you mean to comment this under `NumericQueryResult merge()`? :)

---

Yes, sorry, I did mean to comment under `NumericQueryResult.merge()`. So I'm proposing:

- If `numericQuery()` is invoked for a `QueryService` instance with chunking, then the caller must pass `function='count' buckets=1` or else we throw an exception.
- `NumericQueryResult.merge()` should sum the counts rather than concatenating them.

---

For now, at least, I've opted to simply include a big caveat in the `numericQuery` javadoc but leave the power in the user's hands.  IMO this makes the API much more powerful; if we commit to rebucketing on the client (summing counts rather than concantenating them) then we will expose only a very small fraction of the power of `numericQuery`.